Duplicati/WebserverCore/Middlewares/WebsocketExtensions.cs

```diff
@@ -52,24 +53,40 @@ public static IApplicationBuilder UseNotifications(this IApplicationBuilder app,
 
         var result = await ReceiveAsync();
 
-        while (!result.CloseStatus.HasValue)
+        while (!result?.CloseStatus.HasValue == true)
         {
             result = await ReceiveAsync();
         }
 
-        await webSocket.CloseAsync(
-            result.CloseStatus.Value,
-            result.CloseStatusDescription,
-            CancellationToken.None);
+        if (result?.CloseStatus is not null)
+        {
+            await webSocket.CloseAsync(result.CloseStatus.Value, result.CloseStatusDescription, CancellationToken.None);
+        }
 
         return;
 
-        async Task<WebSocketReceiveResult> ReceiveAsync()
+        async Task<WebSocketReceiveResult?> ReceiveAsync()
         {
-            var receiveResult = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), cancellationToken);
+            WebSocketReceiveResult? receiveResult;
+            try
+            {
+                receiveResult = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), cancellationToken);
+            }
+            catch (WebSocketException e)
+                when (e is
+                      {
+                          Message:
+                          "The remote party closed the WebSocket connection without completing the close handshake."
```

Comment:
Seems a little brittle to rely on the exception message? Can it appear translated for instance?

Reply:
Ok it should be better now. Though I still not sure how to solve this one: 

> Duplicati.WebserverCore.Exceptions.UnauthorizedException: Token family not found
   at Duplicati.WebserverCore.Services.TokenFamilyStore.GetTokenFamily(String userId, String familyId, CancellationToken ct) in /home/natan/git/FOOS/duplicati/Duplicati/WebserverCore/Services/TokenFamilyStore.cs:line 41
   at Duplicati.WebserverCore.Services.LoginProvider.PerformLoginWithRefreshToken(String refreshTokenString, CancellationToken ct) in /home/natan/git/FOOS/duplicati/Duplicati/WebserverCore/Services/LoginProvider.cs:line 29

Reply:
I think while you refreshing browser constantly this is true `if (tokenFamily.Counter != refreshToken.Counter)` and token is removed from DB.

Reply:
Yes, I think you can trigger it with a race where you reload the page after the request has been sent, but before the response is stored in the browser. In this case, the browser will have the previous token. Next request will send the old token which will mismatch with the database and it will clear the tokens.

I think it is a different PR that should solve that, the fix you made looks good for the original task.